test-app/src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/service/MessageListener.kt

```diff
@@ -7,21 +7,19 @@ import org.springframework.stereotype.Service
 @Service
 class InboundMessageListener(private val inboundMessageService: InboundMessageService, private val objectMapper: ObjectMapper) {
 
-  @SqsListener(queueNames = ["inboundqueue"], factory = "hmppsQueueContainerFactoryProxy")
-  fun processMessage(rawMessage: String?) {
-    val (Message) = objectMapper.readValue(rawMessage, Message::class.java)
-    val event = objectMapper.readValue(Message, HmppsEvent::class.java)
+  @SqsListener("inboundqueue", factory = "hmppsQueueContainerFactoryProxy")
+  fun processMessage(message: Message) {
+    val event = objectMapper.readValue(message.Message, HmppsEvent::class.java)
```

Comment:
we _could_ add a custom converter to convert into a specific type as well

Reply:
Sounds like a good idea to me.